server/db/driver/pg/internal/orders.go

```diff
@@ -47,6 +47,23 @@ const (
 		commit, coins, quantity, rate, force, filled
 	FROM %s WHERE status = $1;`
 
+	// CountPreimageMisses returns the number of orders with no preimage for a
+	// given user. When the table is the archived orders, these are misses.
+	CountPreimageMisses = `SELECT count(*)
+	FROM %s
+	WHERE preimage IS NULL
+		AND account_id = $1;`
+
+	CountPreimageStatsLastNOrders = `SELECT COUNT(*)
+	FROM (
+		SELECT preimage IS NULL AS preimageMiss
+		FROM %s -- e.g. dcr_btc.orders_archived
+		WHERE account_id = $1
+		ORDER BY epoch_idx * epoch_dur DESC -- epoch close time
```

Comment:
Sucks we have to do the math here.

Reply:
I actually deciding ages ago to not have matchTime stored explicitly because it's specified by these values.  Perhaps because I wasn't surprised it didn't bother me.  Although now that I talk about this, I realize this product is epoch *start* time!  This should actually be `(epoch_idx+1) * epoch_dur` for epoch close.